pubtools/pulplib/_impl/fake/client.py

```diff
@@ -79,6 +82,35 @@ def get_repository(self, repository_id):
 
         return f_return(data[0])
 
+    def _do_upload_file(self, upload_id, file_obj, name):
+        # pylint: disable=unused-argument
+        out = f_return((hashlib.sha256().hexdigest(), random.randint(10, 1000)))
```

Comment:
Is there a particular reason that you just return a static checksum and random size, instead of actually reading the file object?

I think it would be preferable to truly read the file object for a couple of reasons:

- so that it crashes if the file object can't actually be read - same as the real client would
- so that you get stable behavior with respect to the checksum & size, instead of using something random

Scaling isn't really a concern with the fake client, so I think you can feel free to implement that in a "simple" way, like just reading the data directly in one chunk.

Reply:
yeah, Agree.